torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -991,6 +991,23 @@ class AllocateReuseModifier {
     auto this_tv = alloc_info->alloc_expr->buffer()->as<TensorView>();
     auto reuse_tv = to_reuse->alloc_expr->buffer()->as<TensorView>();
 
+    // Check that either both tv's are vectorized acceses, or neither are.
+    // Vectorized allocations require correct alignment so they can only alias
+    // with other allocations with the right alignment
+    const auto& va = GpuLower::current()->vectorizedAccesses();
+    if ((va.find(this_tv) == va.end()) != (va.find(reuse_tv) == va.end())) {
+      return false;
+    }
+
+    // Shared memory is all aligned to 128 bits, local memory might not be
```

Comment:
Just curious do we already have 128b alignment for all smem or it's in a different PR?

Reply:
I thought we already align it, is it not the case?

Reply:
Looks like TOT aligns to data type size:

[https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/codegen.cpp#L1293-L1308](https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/codegen.cpp#L1293-L1308)

I modified the static version in #1439 .